src/kernel/src/arch/mod.rs

```diff
@@ -51,4 +57,52 @@ impl MachDep {
 
         Ok(SysOut::ZERO)
     }
+
+    pub fn tsc_freq(&self) -> &AtomicU64 {
+        &self.tsc_freq
+    }
+
+    #[cfg(target_arch = "x86_64")]
+    fn init_tsc() -> AtomicU64 {
+        let duration = Duration::from_millis(1);
+        let start = Instant::now();
+
+        let start_tsc = unsafe { std::arch::x86_64::_rdtsc() };
+
+        while Instant::now().duration_since(start) < duration {
+            // Busy-wait
+        }
+
+        let end_tsc = unsafe { std::arch::x86_64::_rdtsc() };
+        let elapsed = end_tsc - start_tsc;
+
+        AtomicU64::new(elapsed * 1000)
+    }
```

Comment:
Did you take this from the PS4 or FreeBSD? What I saw in the PS4 is not like this.

Reply:
FreeBSD.

Reply:
Seems like we have a complicated problem here. The problem is `rdtsc` is not really usablec here because the current thread might get suspended anytime. The best possible solution here should be query this value from the OS.

Reply:
We might need to hard-coded the value here for now. From what I know the only use case for this value is timer system. That mean the worst case for wrong value here is inaccurate timer.

Reply:
So, what should I do with this PR?

Reply:
Ok.